### PR TITLE
code refractoring with separate k8s-client package

### PR DESCRIPTION
**Pull request by rkamudhan**
bug fix for #56 & #57

Change-Id: If4f3db4dd01ca3330d5d36b8a2e4c5740def910a
Co-authored-by: Doug Smith <douglaskippsmith@gmail.com>
Co-authored-by: Tomofumi Hayashi <s1061123@gmail.com>
Co-authored-by: Dan Williams <dcbw@redhat.com>

### CODE REVIEW

File: k8sclient/k8sclient.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+type NoK8sNetworkError string
+
+func (e NoK8sNetworkError) Error() string { return string(e) }
+
+func createK8sClient(kubeconfig string) (*kubernetes.Clientset, error) {
+
+	// uses the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, fmt.Errorf("createK8sClient: failed to get context for the kubeconfig %v, refer Multus README.md for the usage guide", kubeconfig)
```
Comment (s1061123) [inline k8sclient/k8sclient.go]: how about to print err as well as kubeconfig?

File: k8sclient/k8sclient.go [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+	// creates the clientset
+	return kubernetes.NewForConfig(config)
+}
+
+func getPodNetworkAnnotation(client *kubernetes.Clientset, k8sArgs types.K8sArgs) (string, error) {
+	var annot string
+	var err error
+
+	pod, err := client.Pods(string(k8sArgs.K8S_POD_NAMESPACE)).Get(fmt.Sprintf("%s", string(k8sArgs.K8S_POD_NAME)), metav1.GetOptions{})
+	if err != nil {
```
Comment (s1061123) [inline k8sclient/k8sclient.go]: how about to print err in this Errorf?

File: k8sclient/k8sclient.go [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+			}
+
+			podNet = append(podNet, m)
+		}
+	}
+
+	return podNet, nil
+}
+
+func isJSON(str string) bool {
```
Comment (s1061123) [inline k8sclient/k8sclient.go]: now isJSON() seems to used at nowhere, so we could remove it, I guess.

File: k8sclient/k8sclient.go [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+	if networkname == "" {
+		return "", fmt.Errorf("getnetplugin: network name can't be empty")
+	}
+
+	netNsName := "default"
+	if networkinfo["namespace"] != nil {
+		netNsName = networkinfo["namespace"].(string)
+	}
+
+	tprclient := fmt.Sprintf("/apis/kubernetes.cni.cncf.io/v1/namespaces/%s/networks/%s", netNsName, networkname)
```
Comment (s1061123) [inline k8sclient/k8sclient.go]: thank you for your changes to track CRD spec!

**Comment by rkamudhan** [comment]
@s1061123 Will take all the comments and create a issues. Thanks for the approval.

**Pull request merged**